src/Microsoft.Extensions.Logging.Analyzers/Descriptors.cs

```diff
@@ -8,15 +8,23 @@ namespace Microsoft.Extensions.Logging.Analyzers
     internal class Descriptors
     {
         public static DiagnosticDescriptor MEL1NumericsInFormatString = new DiagnosticDescriptor(
-            "MEL1", "Numerics should not be used in logging format string",
+            "MEL0001", "Numerics should not be used in logging format string",
             "Numerics should not be used in logging format string", "Usage", DiagnosticSeverity.Info, true);
 
         public static DiagnosticDescriptor MEL2ConcatenationInFormatString = new DiagnosticDescriptor(
-            "MEL2", "Logging format string should not be dynamically generated",
+            "MEL0002", "Logging format string should not be dynamically generated",
             "Logging format string should not be dynamically generated", "Usage", DiagnosticSeverity.Info, true);
 
         public static DiagnosticDescriptor MEL3FormatParameterCountMismatch = new DiagnosticDescriptor(
-            "MEL3", "Logging format string parameter count mismatch",
+            "MEL0003", "Logging format string parameter count mismatch",
             "Logging format string parameter count mismatch", "Usage", DiagnosticSeverity.Warning, true);
+
+        public static DiagnosticDescriptor MEL4UseCompiledLogMessages = new DiagnosticDescriptor(
```

Comment:
Should we rename properties to be 0003 too?

Reply:
I'll do that. It's most important for the ID since that's a string and is sorted as a string.